gateway/gateway.go

```diff
@@ -124,22 +97,16 @@ type Gateway struct {
 	// Defaults to noop.
 	FatalErrorCallback func(err error)
 
-	// OnScalingRequired is the function called, if Discord closes with error
-	// code 4011 aka Scaling Required. At the point of calling, the Gateway
-	// will be closed, and can, after increasing the number of shards, be
-	// reopened using Open. Reconnect or ReconnectCtx, however, will not be
-	// available as the session is invalidated.
-	OnScalingRequired func()
-
 	// AfterClose is called after each close or pause. It is used mainly for
 	// reconnections or any type of connection interruptions.
 	//
 	// Constructors will use a no-op function by default.
 	AfterClose func(err error)
 
-	waitGroup sync.WaitGroup
+	onScalingRequired func()
```

Comment:
There is no `Scaling Required` error code. It's actually `Sharding Required` and I mistyped it. The original shard manager PR fixes this, and so should this.
```suggestion
	onShardingRequired func()
```